Client/Frontend/Fakespot/Views/ReviewQualityCardView.swift

```diff
@@ -0,0 +1,364 @@
+                                                            size: UX.ratingLetterFontSize,
+                                                            weight: .semibold)
+    }
+
+    private var ratingHeightConstraint: NSLayoutConstraint?
+    private var ratingWidthConstraint: NSLayoutConstraint?
+
+    init(rating: FakespotReliabilityRating) {
+        self.rating = rating
+        super.init(frame: .zero)
+        setupNotifications(forObserver: self,
+                           observing: [.DynamicFontChanged])
+        setupLayout()
+        setupView()
+        reliabilityLetterLabel.text = rating.letter
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupLayout() {
+        let size = min(UIFontMetrics.default.scaledValue(for: UX.ratingSize), UX.maxRatingSize)
+        ratingHeightConstraint = heightAnchor.constraint(equalToConstant: size)
+        ratingHeightConstraint?.isActive = true
+
+        ratingWidthConstraint = widthAnchor.constraint(equalToConstant: size)
+        ratingWidthConstraint?.isActive = true
+
+        addSubview(reliabilityLetterView)
+        reliabilityLetterView.addSubview(reliabilityLetterLabel)
+
+        NSLayoutConstraint.activate([
+            reliabilityLetterView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            reliabilityLetterView.topAnchor.constraint(equalTo: topAnchor),
+            reliabilityLetterView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            reliabilityLetterView.bottomAnchor.constraint(equalTo: bottomAnchor),
+
+            reliabilityLetterLabel.centerXAnchor.constraint(equalTo: reliabilityLetterView.centerXAnchor),
+            reliabilityLetterLabel.centerYAnchor.constraint(equalTo: reliabilityLetterView.centerYAnchor)
+        ])
+        adjustLayout()
+    }
+
+    private func setupView() {
+        layer.cornerRadius = UX.cornerRadius
+        layer.borderWidth = 1
+        clipsToBounds = true
+        layer.backgroundColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1).cgColor
+        layer.borderColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.15).cgColor
+    }
+
+    private func adjustLayout() {
+        ratingHeightConstraint?.constant = min(UIFontMetrics.default.scaledValue(for: UX.ratingSize), UX.maxRatingSize)
+        ratingWidthConstraint?.constant = min(UIFontMetrics.default.scaledValue(for: UX.ratingSize), UX.maxRatingSize)
+
+        setNeedsLayout()
+        layoutIfNeeded()
+    }
+
+    func handleNotifications(_ notification: Notification) {
+        switch notification.name {
+        case .DynamicFontChanged:
+            adjustLayout()
+        default: break
+        }
+    }
+
+    func applyTheme(theme: Theme) {
+        reliabilityLetterView.layer.backgroundColor = rating.color(theme: theme).cgColor
+    }
+}
+
+final class ReviewQualityCardView: UIView, Notifiable, ThemeApplicable {
```

Comment:
The class is not designed with inheritance or subclassing in mind and for preventing unintended subclassing I declared it final. This can also lead to more efficient method dispatch.